scripts/docs-ci-helper.bash

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+set -euo pipefail
+
+function build {
+  if [[ -z "$1" || -z "$2" || -z "$3" || -z "$4" ]]; then
+    echo "Usage: build (book|docs|docs-wasm) REF DIR INVALIDATION_PATH"
+    exit 1
+  fi
+
+  local BUILD_TYPE="$1"
+  if [[ $BUILD_TYPE != "docs" && $BUILD_TYPE != "book" && $BUILD_TYPE != "docs-wasm" ]]; then
+    echo "Available build types: book, docs, docs-wasm"
+    exit 1
+  fi
+
+  local REF="$2"
+  local DIR="$3"
+  local INVALIDATION_PATH="$4"
+
+  # Checkout the ref we were given
+  git checkout -q "$REF"
+  HEAD_REV="$(git rev-parse HEAD)"
+
+  # Check if the existing ref matches
+  if [[ "$(cat "$DIR"/.rev)" = "$HEAD_REV" ]]; then
+    echo "Cached $BUILD_TYPE build for $REF found in $DIR!"
+    exit 0
+  fi
+
+  rm -rf "$DIR"
+  mkdir -p "$DIR"
+  echo "Building $BUILD_TYPE for $REF..."
+
+  case $BUILD_TYPE in
+    "docs")
+      cargo doc --all --features="animation gltf vulkan"
+      mv target/doc/* "$DIR/"
+      ;;
+    "docs-wasm")
+      # TODO: build wasm branch here
+      ;;
+    "book")
+      mdbook build book -d "../$DIR"
+      ;;
+    *)
+      exit 1
+  esac
+
+  # Write the newly built rev to the rev file
+  echo "$HEAD_REV" >> "$DIR/.rev"
+
+  # Write the invalidation path since we just rebuilt
+  echo "$INVALIDATION_PATH" >> ./book-paths-updated
+}
+
+function invalidate_aws {
+  # Check if there are any updated docs paths
+  if [[ -f "docs-paths-updated" ]]; then
```

Comment:
`docs-paths-updated` doesn't appear to be written to anywhere, is this left for later?

Reply:
Thanks, totally missed that!